code/modules/projectiles/guns/magic/staff.dm

```diff
@@ -54,3 +54,108 @@
+	fire_sound = "sound/magic/WandODeath.ogg"
+	ammo_type = /obj/item/ammo_casing/magic/staffofrevenant
+	icon_state = "staffofrevenant"
+	item_state = "staffofrevenant"
+	w_class = 4
+	max_charges = 1
+	recharge_rate = 10
+	no_den_usage = 1
+	var/revenant_level = 0
+	var/revenant_damage = 20 // doesn't actually change the damage, only used for the inhand activation
+	var/revenant_souls = 0
+	var/list/drained_mobs = list() //Cannot harvest the same mob twice
+
+/obj/item/weapon/gun/magic/staff/staffofrevenant/attack(mob/living/carbon/human/target, mob/living/user)
+	if(target.stat & DEAD)
+		if(!(target in drained_mobs))
+			playsound(src,'sound/magic/Staff_Chaos.ogg',40,1)
+			user.visible_message("<font color=purple>[user] drains [target] their soul with [src]!</font>", "<span class='notice'>You use [src] to drain [target]'s soul, empowering your weapon!</span>")
+			revenant_souls++
+			drained_mobs.Add(target)
+		else
+			user << "<span class='warning'>[target]'s soul is dead and empty.</span>"
+			return
+	..()
+
+obj/item/weapon/gun/magic/staff/staffofrevenant/attack_self(mob/living/user)
+	if(revenant_level == 0)
+		if(revenant_souls >= 5)
+			user << "<font color=purple>As you focus on the staff, you witness the crystal eminating a bright shine, before receeding again. The staff hums at an eerie tone, and has managed to become much stronger...</font>"
+			max_charges = 2
+			charges = 2
+			recharge_rate = 9
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level1(src)
+			revenant_level = 1
+			revenant_damage = 25
+	else if(revenant_level == 1)
+		if(revenant_souls >= 10)
+			user << "<font color=purple>Once again, you glance at the staff, sparks now eminating from it as it begins to grow in power. You hear silent wailing around you, as you begin your descent into madness...</font>"
+			max_charges = 3
+			charges = 3
+			recharge_rate = 8
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level2(src)
+			revenant_level = 2
+			revenant_damage = 30
+			user << 'sound/spookoween/ghost_whisper.ogg'
+	else if(revenant_level == 2)
+		if(revenant_souls >= 15)
+			user << "<font color=purple>You only give a quick glimpse at the staff, as you hear the screams of the fallen emenating from the staff's crystal. Your powers grow even stronger...</font>"
+			max_charges = 4
+			charges = 4
+			recharge_rate = 7
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level3(src)
+			revenant_level = 3
+			revenant_damage = 35
+			user << 'sound/hallucinations/veryfar_noise.ogg'
+	else if(revenant_level == 3)
+		if(revenant_souls >= 20)
+			user << "<font color=purple>You only give a quick glimpse at the staff, as you hear the screams of the fallen emenating from the crystal mounted ontop of the staff, echoing throughout the station. Your powers grows even stronger...</font>"
```

Comment:
And also "Your powers grows even stronger..." should be "Your powers **grow** even stronger..."